src/hello-bitcoin/HelloBitcoin.sol

```diff
@@ -0,0 +1,299 @@
+     * The `nextOrderId` is incremented each time a new BTC to USDT swap order is created,
+     * ensuring that each order has a unique identifier.
+     */
+    uint256 nextOrderId;
+
+    /**
+     * @dev Counter for generating unique identifiers for ordinal sell orders.
+     * The `nextOrdinalId` is incremented each time a new ordinal sell order is created,
+     * ensuring that each ordinal order has a unique identifier.
+     */
+    uint256 nextOrdinalId;
+
+    /**
+     * @dev Struct representing a BTC to USDT swap order.
+     */
+    struct BtcSellOrder {
+        uint256 sellAmountBtc; // Amount of BTC to be sold in the order.
+        uint256 buyAmountUsdt; // Amount of USDT to be bought in the order.
+        address btcSeller; // Address of the seller initiating the order.
+        BitcoinAddress btcBuyer; // Bitcoin address of the buyer (initialized with an empty scriptPubKey).
+        bool isOrderAccepted; // Flag indicating whether the order has been accepted.
+    }
+
+    /**
+     * @dev Struct representing an ordinal sell order for swapping BTC to USDT.
+     */
+    struct OrdinalSellOrder {
+        OrdinalId ordinalID; // Unique identifier for the ordinal sell order.
+        uint256 buyAmountUsdt; // Amount of USDT to be bought in the order.
+        BitcoinTx.UTXO utxo; // UTXO associated with the BTC to USDT swap order.
+        address ordinalSeller; // Address of the seller initiating the ordinal order.
+        bool isOrderAccepted; // Flag indicating whether the ordinal order has been accepted.
+        BitcoinAddress ordinalBuyer; // Bitcoin address of the buyer (initialized with an empty scriptPubKey).
+    }
+
+    /**
+     * @dev Struct representing a unique identifier for an ordinal sell order.
+     */
+    struct OrdinalId {
+        bytes32 txId; // Transaction ID associated with the ordinal order.
+        uint32 index; // Index associated with the ordinal order.
+    }
+
+    /**
+     * @dev Struct representing a Bitcoin address with a scriptPubKey.
+     */
+    struct BitcoinAddress {
+        bytes scriptPubKey; // Script public key associated with the Bitcoin address.
+    }
+
+    event swapBtcForUsdtEvent(uint256 indexed orderId, uint256 sellAmountBtc, uint256 buyAmountUsdt);
+    event acceptBtcToUsdtSwapEvent(uint256 indexed id, BitcoinAddress bitcoinAddress);
+    event proofBtcSendtoDestinationEvent(uint256 id);
+
+    event swapOrdinalToUsdtEvent(uint256 indexed id, OrdinalId ordinalID, uint256 buyAmountUsdt);
+    event acceptOrdinalToUsdtSwapEvent(uint256 indexed id, BitcoinAddress bitcoinAddress);
+    event proofOrdinalSellOrderEvent(uint256 id);
+
+    BridgeState.Storage internal relay;
+    TestLightRelay internal testLightRelay;
+
+    /**
+     * @dev Constructor to initialize the contract with the relay and ERC2771 forwarder.
+     * @param _relay The relay contract implementing the IRelay interface.
```

Comment:
Would be nice to add some link to the docs here to give the user guidance on the relay.